docs/secrets/Secrets.md

```diff
@@ -1,98 +1,217 @@
+- Any code running in additional containers that have the service account token mounted.
+
+To prevent access to secrets, untrusted code must be executed in containers where the service account token will not be supplied to (mounting of service account token disabled via pod spec and token not passed into the container in any other way).
+
+More information on how to create image pull secrets in Kubernetes can be found in the Kubernetes documentation:
+
+- [Pull an Image from a Private Registry][k8s_docs-pull_image_private_registry]
+- [Add ImagePullSecrets to a service account][k8s_docs_add_imagepullsecrets_to_service_account]
+
+
+## Git Server Authentication
+
+### Pipeline Clone Secret
+
+Before Steward can run a pipeline, it needs to fetch the pipeline definition (Jenkinsfile) from a Git repository.
+If cloning this Git repository requires authentication, a secret must be provided by the respective Steward client (on behalf of end users).
+
+A Steward PipelineRun resource object can specify the secret to be used to authenticate at the Git server from where the pipeline definition should be taken:
+
+```yaml
+apiVersion: steward.sap.com/v1alpha1
+kind: Tenant
+spec:
+    ...
+    jenkinsFile:
+        repoUrl: https://github.com/org1/pipelines
+        ...
+        repoAuthSecret: github-com-token1
+```
+
+If authentication is not required when cloning the pipeline repository, `spec.jenkinsFile.repoAuthSecret` can be omitted or set to an empty string value.
+
+The value of `spec.jenkinsFile.repoAuthSecret` is the name of a Kubernetes `v1/Secret` resource object of type `kubernetes.io/basic-auth` that contains the username and password for authentication when cloning from `spec.jenkinsFile.repoUrl`.
+Besides that there are no further requirements like special annotations or labels.
+
+When a pipeline gets executed in a transient sandbox namespace, the pipeline clone secret specified in `spec.jenkinsFile.repoAuthSecret` of the corresponding PipelineRun resource object is copied to the sandbox namespace with a different name.
+The Jenkinsfile Runner container has a generated Git credential file (`$HOME/.git-credentials`) that configures the username and password from that secret for the respective Git server.
+This means that any further Git commands executed in the Jenkinsfile Runner container will use these credentials (for the respective Git server) if not explicitly overridden.
+
+__:warning: Warning:__ Any code that gets executed by a pipeline AND runs in the Jenkinsfile Runner container or has access to the Kubernetes service account token can read the pipeline clone secret!
+This is especially important to consider if untrusted code may get executed, e.g. a pipeline processing pull requests from untrusted users.
+
+- Any code running in the Jenkinsfile Runner container, because this container has the credentials in `$HOME/.git-credentials` and has the service account token mounted.
```

Comment:
same here. see above

Reply:
PS: Didn't we want to delete the credential from the container once the pipeline had been fetched?
Note: This would contradict with the "pipeline definition (Jenkinsfile) and the sources are located in the same repository" comment below.

Reply:
> PS: Didn't we want to delete the credential from the container once the pipeline had been fetched?

Yes, but it cannot be done by the Run Controller, but has to be performed by the JFR container entry point. It's is not implemented yet. If we implement it, we can adapt the documentation here.

> Note: This would contradict with the "pipeline definition (Jenkinsfile) and the sources are located in the same repository" comment below.

Yes. If we remove the Secret object and the `.git-credentials` file, then this special case does not apply anymore and would have to be remove in the documentation.